internal/contour/route.go

```diff
@@ -234,12 +230,39 @@ func prefixmatch(prefix string) route.RouteMatch {
 
 // action computes the cluster route action, a *route.Route_route for the
 // supplied ingress and backend.
-func actionroute(namespace, name string, port int, ws bool, timeout time.Duration) *route.Route_Route {
-	cluster := hashname(60, namespace, name, strconv.Itoa(port))
+func actionroute(services []*dag.Service, ws bool, timeout time.Duration) *route.Route_Route {
```

Comment:
Please add a test for actionroute, it's not just a helper for constructing some annoying grpc data object any more, the totalWeight business logic needs a test.